### DISCUSSION TITLE
Nuget.Versioning: Wrong documentation or parsing behaviour.

### DISCUSSION THREAD

**Discussion by HHHackGit** [NuGet/Home#12050] [General]
The [documentation](https://docs.microsoft.com/en-us/nuget/concepts/package-versioning) lists the following table:
<html>
<body>

Notation | Applied rule | Description
-- | -- | --
1.0 | x ≥ 1.0 | Minimum version, inclusive
(1.0,) | x > 1.0 | Minimum version, exclusive
[1.0] | x == 1.0 | Exact version match
(,1.0] | x ≤ 1.0 | Maximum version, inclusive
(,1.0) | x < 1.0 | Maximum version, exclusive
[1.0,2.0] | 1.0 ≤ x ≤ 2.0 | Exact range, inclusive
(1.0,2.0) | 1.0 < x < 2.0 | Exact range, exclusive
[1.0,2.0) | 1.0 ≤ x < 2.0 | Mixed inclusive minimum and exclusive maximum version
(1.0) | invalid | invalid

</body>
</html>

but in the nuget package Nuget.Versioning in the method VersionRange.Parse() another description:

// Summary:
//     The version string is either a simple version or an arithmetic range e.g. 
1.0 --> 1.0 ≤ x
(,1.0] --> x ≤ 1.0
(,1.0) --> x < 1.0
[1.0] --> x == 1.0
(1.0,) --> 1.0 < x 
(1.0, 2.0) --> 1.0 < x < 2.0
[1.0, 2.0] --> 1.0 ≤ x ≤ 2.0

Here you can see the difference in behavior for "Minimum version, inclusive"
In documantation 1.0 -> x ≥ 1.0
In package 1.0 -> 1.0 ≤ x

Which option is correct?

**Comment by zivkan** [comment]
> Here you can see the difference in behavior for "Minimum version, inclusive"
> In documantation 1.0 -> x ≥ 1.0
> In package 1.0 -> 1.0 ≤ x

To me they're both mathematically equivalent. I studied maths far too long ago to remember what it's called, but it's the same equation, just manipulated to show things in a different order, in the same way that `a = b` and `a - b = 0` are equivalent equations.
